api/v1alpha1/spire_oidc_discovery_provider_types.go

```diff
@@ -40,19 +40,31 @@ type SpireOIDCDiscoveryProviderSpec struct {
 	LogFormat string `json:"logFormat,omitempty"`
 
 	// trustDomain to be used for the SPIFFE identifiers
+	// Must be a valid SPIFFE trust domain (lowercase alphanumeric, hyphens, and dots).
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=255
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([a-z0-9\-\.]*[a-z0-9])?$`
 	TrustDomain string `json:"trustDomain,omitempty"`
 
 	// agentSocketName is the name of the agent socket.
+	// Must be a relative file name (no path traversal or absolute paths).
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9._-]+$`
 	// +kubebuilder:default:="spire-agent.sock"
 	AgentSocketName string `json:"agentSocketName,omitempty"`
 
 	// jwtIssuer is the JWT issuer url.
+	// Must be a valid HTTPS or HTTP URL.
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=512
+	// +kubebuilder:validation:Pattern=`^https?://`
 	JwtIssuer string `json:"jwtIssuer,omitempty"`
 
 	// replicaCount is the number of replicas for the OIDC provider.
+	// Must be between 1 and 5.
 	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=5
```

Comment:
The maximum limit for now is subjective, can you please create a separate ticket for performance testing the limits?